src/liballoc/vec.rs

```diff
@@ -1542,146 +1542,26 @@ impl<T: Hash> Hash for Vec<T> {
 
 #[stable(feature = "rust1", since = "1.0.0")]
 #[rustc_on_unimplemented = "vector indices are of type `usize` or ranges of `usize`"]
-impl<T> Index<usize> for Vec<T> {
-    type Output = T;
+impl<T, I> Index<I> for Vec<T> where [T]: Index<I> {
```

Comment:
Can we change these bounds to be the same as those for `[T]` rather than delegating to T's impl explicitly?

Reply:
Are you saying you'd prefer something like this?

```rust
impl<T, I> IndexMut<I> for Vec<T>
where
    I: ::core::slice::SliceIndex<[T]>,
{
    #[inline]
    fn index_mut(&mut self, index: I) -> &mut Self::Output {
        IndexMut::index_mut(&mut **self, index)
    }
}
```

Reply:
@fintelia indeed!

Reply:
Why? The one currently in the PR means that if a crate implements `Index<TheirType> for [T]` they automatically get the impl for `Vec<T>` too, which seems convenient.

Reply:
It's a weird layer of double-indirection that isn't used very commonly in trait bounds. It is apparently different enough that specialization treats it differently as that one regression indicates.